.github/workflows/ci.yml

```diff
@@ -0,0 +1,85 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  go-lint:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    name: Go Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        id: setup-go
+        uses: ./.github/actions/setup-go
+        with:
+          cache-prefix: go-lint
+
+      - name: Compute tools cache info
+        id: tools-cache-info
+        run: |
+          echo path=$(go env GOPATH)/bin >> $GITHUB_OUTPUT
+          echo make-hash=$(make -n install-tools | sha256sum | cut -d' ' -f1) >> $GITHUB_OUTPUT
+
+      - name: Setup tools cache
+        uses: actions/cache@v4
+        id: tools-cache
+        with:
+          path: ${{ steps.tools-cache-info.outputs.path }}
+          key: tools-go-${{ steps.setup-go.outputs.go-version }}-make-${{ steps.tools-cache-info.outputs.make-hash }}
+
+      - name: Install tools
+        if: steps.tools-cache.outputs.cache-hit != 'true'
+        env:
+          GOCACHE: /tmp/tools/go-build
+          GOMODCACHE: /tmp/tools/go-mod
+        run: make install-tools
+
+      - name: Check formatting
+        run: |
+          make go-format
```

Comment:
actually, we can remove `gofumpt` completely by removing this target from the makefile, and checking what (if any) files are modified - just use the standard Go built-in command `gofmt` with its `-l` option - it will print out a list of any files that need formatting, so you can do `go fmt -l . | wc -l` and if it's not zero, the format-check should consider it an error

Reply:
`gofmt` should be included with any standard distribution of the go compiler and toolset

Reply:
Hi @richscott . Please take a look now if the changes are ok